I4 markdown exercises

JulianKarlBauer opened the pull request:
I think there is slightly too much text inside this exercise... What do you think?

1. Is including clickable bare URL, e.g. <https://carpentries.org/> beneficial?
2. Do you think the blanks are well-placed?

JulianKarlBauer committed:
Use reference links and _includes/links.md

Do what we teach ;)

JulianKarlBauer committed:
Exercise on links: <>, [](), [][]

JulianKarlBauer committed:
Merge branch 'gh-pages' into i4_markdown_exercises

JulianKarlBauer committed:
Only use one example URL: https://carpentries.org/

JulianKarlBauer committed:
Remove "link" to non-existing file

JulianKarlBauer committed:
Simplify excercise and move motivation and <bare links> outside the exc.

sstevens2 commented on _episodes/02-markdown.md:
```diff
@@ -221,8 +221,46 @@ keypoints:
 
+Linking other websites is an essential feature of a website.
+Bare URLs like https://carpentries.org/ can be made clickable by wrapping them into 
+angle brackets like this <https://carpentries.org/>.
```
I think you might need to escape the brackets with \< and \> if you want to see them in the text instead of have them create the link.

JulianKarlBauer replied:
Thanks! I didn't think of this and tried to be more specific in 199da9be92be5f7f83abe089259750b779d6794e

sstevens2 commented on _episodes/02-markdown.md:
```diff
@@ -221,8 +221,46 @@ keypoints:
+> ## Links
+> Fill in the blanks below to create this rendered view
+> 
+> Examples:
+> 1. [Inline link](https://carpentries.org/)
+> 2. [Reference link][case-InSeNsiTiVe-reference-tag]
+>
+> [case-insensitive-reference-tag]: https://carpentries.org/
```
I'm confused about the purpose of this section?  It seems to be repeated in the solution?

JulianKarlBauer replied:
Thank for looking at this!

My intention is to have these three stages:
1. Rendered
2. Fill-in-the-blanks
3. Source

![image](https://user-images.githubusercontent.com/55383008/91875742-335c2c80-ec7c-11ea-91a3-0deb71295b6b.png)

sstevens2 replied:
That makes sense! Apologies for my confusion

sstevens2 commented on _episodes/02-markdown.md:
```diff
@@ -221,8 +221,46 @@ keypoints:
+> {: .source }
+>
+> > ## Solution
+> > ~~~
+> > Examples:
+> > 1. [Inline link](https://carpentries.org/)
+> > 2. [Reference link][case-InSeNsiTiVe-reference-tag]
+> >
+> > [case-insensitive-reference-tag]: https://carpentries.org/
```
~Do these get rendered as links in the lesson or does the code black keep them as plain text?~  One of the complications of teaching markdown in a markdown lesson. ;)

sstevens2 replied:
Ignore this comment.  the first exercise in this lesson seems to be fine so it looks like it won't render the md inside a code block.

sstevens2 reviewed (commented):
I like the addition of the bare URL info.  Left a few comments below about showing the formatting vs it getting rendered inside of the lesson.  Not sure quite what you are asking in 2 above?  The extra blank lines?

annefou commented:
Just a question (and it mostly relates to the next episode on github). Will learners create markdown file locally or will they already create a github repository?

sstevens2 commented:
@annefou I think they should make it in github.  It they don't have to do make it a gh-pages site at this point in the lesson though.

annefou commented:
@sstevens2 I agree so I guess a section on the creation of the github repo will be added at some point. In the next episode (3 github) I started from an existing github repository with index.md file.

sstevens2 commented:
@annefou good point!  maybe when we write the full markdown lesson we should have them create a repo at the beginning?  or add an episode to setup a github repo before? and then they try out markdown in a file in that repo?

annefou commented:
I would probably add it at the beginning of the episode. Otherwise it will be hard for learners to see what they write in markdown. Or we need to user another external tool such as hackMD. Probably better to create a repo.

JulianKarlBauer committed:
Make usage of <> more explicit, thanks Sarah!

tobyhodges committed:
add lines to mark out start of rendered exercise content

tobyhodges commented:
@JulianKarlBauer in 1d006da I added lines to indicate the start of the rendered view the learner should try to recreate in each exercise. I think this improves the readability a little but don't feel all that strongly. If you'd prefer, I'll be happy to revert the change. Let me know, and then this PR is ready to be merged.

tobyhodges commented on _includes/links.md:
```diff
@@ -0,0 +1,3 @@
+[github-flavored-markdown]: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet
+[gitlab-flavored-markdown]: https://docs.gitlab.com/ee/user/markdown.html
+
```
Since we moved over to the remote theme (see #5), the `_includes/link.md` file has been removed. I think it would be good to still use that approach in this lesson, which means we need to replicate the full set of link refs from https://github.com/carpentries/carpentries-theme here. That will allow the theme to use the refs in our local `links.md` file instead of the remote equivalent. If some of that didn't make sense don't worry, the solution is simple enough: add the entire contents of [this file](https://github.com/carpentries/carpentries-theme/blob/master/_includes/links.md) to the `_includes/links.md` file in your branch.

JulianKarlBauer replied:
Thank you, I copied the links from bf45de8.

JulianKarlBauer committed:
Add links from 
https://github.com/carpentries/carpentries-theme/blob/master/_includes/links.md

following instruction

https://github.com/carpentries-incubator/building-websites-with-jekyll-and-github-or-gitlab/pull/26#discussion_r484929045
